File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.java
class Solution {
    public int[] solution(String[] keyinput, int[] board) {
        // up [0, 1] down [0, -1] left [-1, 0] right [1, 0]
        
        int[] answer = new int[board.length];
        
        // 맵의 크기에 따라 이동할 칸이 초과하지 않는지 확인
        int x = board[0] / 2;
        int y = board[1] / 2;
        
        // 입력한 방향키의 배열 요소와 같은 문자에 따라 좌표 이동
        // down, left 경우 음수가 될 수 있으므로 절대값 반환 후 비교
        for (int i = 0; i < keyinput.length; i++) {
            if (keyinput[i].equals("up")) {
                answer[1]++; // [0, 1]
                answer[1] = (y < answer[1]) ? answer[1] - 1 : answer[1];
            }
            
            if (keyinput[i].equals("down")) {
                answer[1]--; // [0, -1]
                answer[1] = (y < Math.abs(answer[1])) ? answer[1] + 1 : answer[1];
            }
            
            if (keyinput[i].equals("left")) {
                answer[0]--; // [-1, 0]
                answer[0] = (x < Math.abs(answer[0])) ? answer[0] + 1 : answer[0];
            }
            
            if (keyinput[i].equals("right")) {
                answer[0]++; // [1, 0]
                answer[0] = (x < answer[0]) ? answer[0] - 1 : answer[0];
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.js
function solution(keyinput, board) {
    // up [0, 1] down [0, -1] left [-1, 0] right [1, 0]
    
    
    // 맵의 크기에 따라 이동할 칸이 초과하지 않는지 확인
    // 소수점 버려야 비교 가능
    var x = Math.floor(board[0] / 2);
    var y = Math.floor(board[1] / 2);
    var answer = [0, 0];
    
    // 입력한 방향키의 배열 요소와 같은 문자에 따라 좌표 이동
    // Math.abs() 사용 시 down, left 음수 이동이 제한됨
    for (let i = 0; i < keyinput.length; i++) {
        if (keyinput[i] === "up" && answer[1] < y) {
            answer[1]++;
        } else if (keyinput[i] === "down" && -y < answer[1]) {
            answer[1]--;
        } else if (keyinput[i] === "left" && -x < answer[0]) {
            answer[0]--;
        } else if (keyinput[i] === "right" && answer[0] < x) {
            answer[0]++;
        }
    }
    
    return answer;
}